### PR TITLE
fix issues in templates, add support for dynamic logical network, usi…

### PR DESCRIPTION
**Pull request by kumdeepakgit**
…ng different types of image (marketplace, galleryimage) to provision the VM, added support for proxyconfig in VM, refactor

# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

**Review by kalvarahul — APPROVED** [review]

File: quickstarts/microsoft.azurestackhci/vm-windows-disks-and-adjoin/prereqs/logical-network/main.bicep [hunk 1]
```diff
@@ -1,47 +1,65 @@
...
-param dnsServers array = [
-  '1.1.1.1'
-  '8.8.8.8'
-]
+param dnsServers array = []
 @description('The VLAN ID for the logical network. If not specified, the default value is 0.')
 param vlan int = 0
 @description('The address prefix for the logical network - for example: 172.16.0.0/22')
-param addressPrefix string
+param addressPrefix string = ''
```
Comment (mbrat2005) [inline quickstarts/microsoft.azurestackhci/vm-windows-disks-and-adjoin/prereqs/logical-network/main.bicep:11]: Make this an optional parameter with a default null value?

```suggestion
param addressPrefix string?
```

File: quickstarts/microsoft.azurestackhci/vm-windows-disks-and-adjoin/prereqs/logical-network/main.bicep [hunk 2]
```diff
@@ -1,47 +1,65 @@
...
+@description('IP address allocation method and could be Static or Dynamic.')
+@allowed([
+  'Static'
+  'Dynamic'
+])
+param ipAllocationMethod string = 'Dynamic'
 @description('The default gateway for the logical network - for example: 172.16.0.1')
-param defaultGateway string
+param defaultGateway string = ''
```
Comment (mbrat2005) [inline quickstarts/microsoft.azurestackhci/vm-windows-disks-and-adjoin/prereqs/logical-network/main.bicep:19]:
```suggestion
param defaultGateway string?
```

File: quickstarts/microsoft.azurestackhci/vm-windows-disks-and-adjoin/prereqs/logical-network/main.bicep [hunk 3]
```diff
@@ -1,47 +1,65 @@
...
 @description('The start IP address for the IP pool - for example: 172.16.1.100')
-param startIPAddress string
+param startIPAddress string = ''
```
Comment (mbrat2005) [inline quickstarts/microsoft.azurestackhci/vm-windows-disks-and-adjoin/prereqs/logical-network/main.bicep:21]:
```suggestion
param startIPAddress string?
```

File: quickstarts/microsoft.azurestackhci/vm-windows-disks-and-adjoin/prereqs/logical-network/main.bicep [hunk 4]
```diff
@@ -1,47 +1,65 @@
...
 @description('The start IP address for the IP pool - for example: 172.16.1.200')
-param endIPAddress string
+param endIPAddress string = ''
```
Comment (mbrat2005) [inline quickstarts/microsoft.azurestackhci/vm-windows-disks-and-adjoin/prereqs/logical-network/main.bicep:23]:
```suggestion
param endIPAddress string?
```

**Comment by kumdeepakgit** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Pull request merged**